src/hooks/useTokenizeWorker.js

```diff
@@ -0,0 +1,93 @@
+import {useState, useRef, useEffect} from 'react';
+import shallowEquals from 'shallow-equal/objects';
+import arrayShallowEquals from 'shallow-equal/arrays';
+import {flatMap} from 'lodash';
+
+const uid = (() => {
+    let current = 0;
+
+    return () => {
+        current = current + 1;
+        return current;
+    };
+})();
+
+// This is actually a hack around the lack of custom comparator support in `useEffect` hook.
+const useCustomEqualIdentifier = (value, equals) => {
```

Comment:
Ugly but necessary hack 😭

Reply:
Do you have a plan to add a dependency

Reply:
What should be listed in dependency? I assume `value` itself is the only dependency, `equals` should never change

Reply:
I'm implying `rehooks`